source/reference/schema.rst

```diff
@@ -1,39 +1,160 @@
+.. index::
+   single: $schema
+
+Declaring a Specification Version
+=================================
+
+.. contents:: :local:
+
+Not only have there have been multiple versions of the specification,
+but it's also possible to add your own extensions to create your own
+version. JSON Schema provides ways for you to declare which version
+a schema conforms to and provides ways to describe your custom
+keywords.
+
 .. index::
    single: $schema
    single: schema; keyword
 
 .. _schema:
 
-The $schema keyword
-===================
+$schema
+-------
 
-.. contents:: :local:
+The ``$schema`` keyword is used to declare which version of the JSON
+Schema specification the schema was written for. The value of the
+``$schema`` keyword is also the identifier for a schema that can be
+used to verify that the schema is valid according to the specification
+version ``$schema`` identifies. A schema that describes another schema
+is called a "meta-schema".
+
+``$schema`` applies to the entire document and must be at the root
+level. It does not apply to externally referenced (``$ref``)
+documents. Those schemas need to declare their own ``$schema``.
+
+If ``$schema`` is not used, an implementation might allow you to
+specify a value externally or it might make assumptions about which
+specification version should be used to evaluate the schema. It's
+recommended that all JSON Schemas have a ``$schema`` entry to
+communicate to readers and tooling which specification version is
+intended. Therefore most of the time, you'll want this at the root of
+your schema::
+
+    "$schema": "http://json-schema.org/draft-07/schema#"
+
+.. draft_specific::
+
+    --Draft-4
+    The identifier for Draft 4 is ``http://json-schema.org/draft-04/schema#``.
+
+    Draft 4 defined a value for ``$schema`` without a specific version
+    (``http://json-schema.org/schema#``) which meant, use the latest
+    version. This has since been deprecated and should no longer be
+    used.
+
+    You might come across references to Draft 5. There is no Draft 5
+    release of JSON Schema. Draft 5 refers to a no-change revision of
+    the Draft 4 release. It does not add, remove, or change any
+    functionality. It only updates references, makes clarifications,
+    and fixes bugs. Draft 5 describes the Draft 4 release. If you came
+    here looking for information about Draft 5, you'll find it under
+    Draft 4. We no longer use the "draft" terminology for this kind of
+    specification update to avoid this confusion.
```

Comment:
I think you mean, we no longer use numerical drafts as the identifier for the draft release now, right? We do still call them "draft YYYY-MM".

Reply:
That's not what I meant at all. I meant we don't refer to a patch release as a "draft". The release designation of the draft-07 patch release is the same as the original release: draft-07. The IETF designations are also called drafts, but those are always different: draft-handrews-json-schema-00 and draft-handrews-json-schema-01. A patch release is a new IETF "draft", but it's not a new JSON Schema "draft".

I'll try again :cry:

Reply:
@Relequestual I made a small change. Hopefully it helps. I'm really not sure how else to say this concisely and I don't think it would be the right thing to go into a full explanation of how IETF and I-Ds work and how we got to where we are now.